sources/presentation/Xenko.Core.Presentation/ValueConverters/ObjectToTypeName.cs

```diff
@@ -17,13 +22,87 @@ public class ObjectToTypeName : OneWayValueConverter<ObjectToTypeName>
         /// <summary>
         /// The string representation of the type of a null object
         /// </summary>
-        public const string NullObjectType = "(None)";
+        public const string NullObjectType = "(None)";        
 
         /// <inheritdoc/>
         [NotNull]
         public override object Convert(object value, Type targetType, object parameter, CultureInfo culture)
         {
-            return value?.GetType().Name ?? NullObjectType;
+            if (value == null) return NullObjectType;
+
+            return value.GetType().ToSimpleCSharpName();
+            
+        }
+    }
+
+
+    /// <summary>
+    /// Helper class to get formatted names from <see cref="Type"/>.
+    /// </summary>
+    public static class TypeNameHelper
```

Comment:
Either it should be a private nested type of ObjectToTypeName class, or be in its own separate file.

Reply:
I mostly made it a separate public class for unit test purposes. If we keep it public I guess I'll need to find a good home for it. It is a fairly generic helper class so could possibly be used elsewhere. 🤔

Reply:
Maybe I'll put it with the other extensions in the `Xenko.Core.Presentation.Extensions` namespace.

Reply:
You can make it internal, and add `InternalsVisibleTo`  attribute in `AssemblyInfo.cs` for unit test purpose.

Reply:
OK, I'll move it to a separate file in `Xenko.Core.Presentation.Extensions`, and make it internal etc.

Reply:
I have moved `TypeNameHelper` to it's own file in `Xenko.Core.Presentation.Extensions` and renamed it to `TypeExtensions` to make it more inline with the rest of the classes in that namespace.  I also moved it's tests to it's own unit test class.